daily-rotate-file.js

```diff
@@ -121,11 +121,13 @@ if (semver.major(winston.version) === 2) {
     };
 }
 
-DailyRotateFile.prototype._final = function (callback) {
-    this.logStream.end(function () {
-        callback = callback || noop;
-        callback();
-    });
+DailyRotateFile.prototype.close = function () {
```

Comment:
I'm a little concerned about this function name. Streams already expose a `close` event and this feels ripe for confusion. For my part (not a stream expert by any stretch), I had to go and look at the core NodeJS `stream` and `winston-transport` source code to verify that this wasn't overwriting a pre-existing `close()` function.

If this is specifically for `winston-daily-rotate-rotate` (WDRF) would it be possible to sidestep any such confusion and call it something WDRF-specific? Something like `endRotation()` or `endLogging()` or `closeTransport()`?

@indexzero This raises an interesting question. Is there a set of functions (hooks?) defined in `winston-transport` that cover setup and teardown? Or at least a convention by which Transports can name basic functionality like this?

Reply:
@ericdrobinson 

>For my part (not a stream expert by any stretch), I had to go and look at the core NodeJS stream and winston-transport source code to verify that this wasn't overwriting a pre-existing close() function.

FWIW, I did the same thing.  

However, I think it does have meaning to `winston`.  In the area I explored with the rotation bug --   In `winston-transport` -- the `unpipe` handler calls close on the `Transport` if it is defined: (https://github.com/winstonjs/winston-transport/blob/master/index.js#L49-L62)

Perhaps this just needs some clarity in the documentation -- maybe this is also the "most correct" place for transports to call `this.end()" and do additional cleanup if necessary since `_final()` is not an option in node versions < 8.0. I'll add more thoughts about that to the thread about `_final()`.